garrysmod/lua/includes/modules/cvars.lua

```diff
@@ -18,111 +11,124 @@ module( "cvars" )
 
 local ConVars = {}
 
-function GetConVarCallbacks( name, CreateIfNotFound )
-
-	local Tab = ConVars[ name ]
+--[[---------------------------------------------------------
+	Name: GetConVarCallbacks
+	Desc: Returns a table of the given ConVars callbacks
+-----------------------------------------------------------]]
+function GetConVarCallbacks( name, createIfNotFound )
 
-	if ( CreateIfNotFound && !Tab ) then
-		Tab = {}
-		ConVars[ name ] = Tab
+	local tab = ConVars[ name ]
+	if ( createIfNotFound and !tab ) then
+		tab = {}; ConVars[ name ] = tab
 	end
 
-	return Tab
+	return tab
+
 end
 
 --[[---------------------------------------------------------
 	Name: OnConVarChanged
 	Desc: Called by the engine
 -----------------------------------------------------------]]
-function OnConVarChanged( name, oldvalue, newvalue )
+function OnConVarChanged( name, old, new )
 
-	local Callbacks = GetConVarCallbacks( name )
-	if ( !Callbacks ) then return end
+	local tab = GetConVarCallbacks( name )
+	if ( !tab ) then return end
 
-	for k, v in pairs( Callbacks ) do
-
-		if ( istable( v ) ) then
-			v[ 1 ]( name, oldvalue, newvalue )
+	for i = 1, #tab do
+		local callback = tab[ i ]
+		if ( type( callback ) == "table" ) then
+			callback[ 1 ]( name, old, new )
 		else
-			v( name, oldvalue, newvalue )
+			callback( name, old, new )
 		end
-
 	end
 
 end
 
 --[[---------------------------------------------------------
-	Name: OnConvarChanged
-	Desc: Called by the engine
+	Name: AddChangeCallback
+	Desc: Adds a callback to be called when convar changes
 -----------------------------------------------------------]]
-function AddChangeCallback( name, func, sIdentifier )
-
-	if ( sIdentifier ) then
-		assert( isstring( sIdentifier ), format( "bad argument #%i (string expected, got %s)", 3, type( sIdentifier ) ) )
-	end

```

Comment:
What is the point of removing these? This is a regression that will cause more confusion for the people incorrectly using these functions during development.

Reply:
what is the point of this restriction, what confusion are we talking about if it doesn't affect anything, in fact the identifier can be really anything, there are no operations with string

Reply:
This value can be `nil` altogether isn't that confusing? Why a value that should be a string can be `nil`, how we can find and remove this value?

Reply:
Returned